File: ios/CustomModule.swift
import Foundation

@objc(CustomModule)
class CustomModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func sendMessage(_ message: String, callback: RCTResponseSenderBlock) {
    let response = "Received: \(message)"
    callback([response])
  }
}

File: android/app/src/main/java/com/authnativebridge/CustomModule.kt
package com.authnativebridge

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Callback

class CustomModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    override fun getName(): String {
        return "CustomModule"
    }

    @ReactMethod
    fun sendMessage(message: String, callback: Callback) {
        val response = "Received: $message"
        callback.invoke(response)
    }
}